src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectManagementService.cs

```diff
@@ -4,14 +4,13 @@
 using System.Collections.Generic;
 using System.Composition;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 using EnvDTE;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.CodeAnalysis.ProjectManagement;
 using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
 using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.Extensions;
+using Microsoft.VisualStudio.Progression;
```

Comment:
I had to go look up UIThread.Invoke's implementation in the Progression sources. I would rather us use platform features such as ThreadHelper.Generic.Invoke or the JoinableTaskFactory (JTF).